libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/reposerver/ReposerverClient.scala

```diff
@@ -60,11 +62,11 @@ trait ReposerverClient {
 
   def createRoot(namespace: Namespace, keyType: KeyType): Future[RepoId]
 
-  def fetchRoot(namespace: Namespace): Future[SignedPayload[RootRole]]
+  def fetchRoot(namespace: Namespace): Future[(RepoId, SignedPayload[RootRole])]
 
   def repoExists(namespace: Namespace)(implicit ec: ExecutionContext): Future[Boolean] =
     fetchRoot(namespace).transform {
-      case Success(_) | Failure(KeysNotReady) => Success(true)
+      case Success(_) | Failure(KeysNotReady) | Failure(MissingRepoIdHeader) => Success(true)
```

Comment:
Why does this mean the repo exists? It's a bad request.. not sure it's a good idea to treat this as a success.

Reply:
Because the root.json was found, even though the repoId was not present (but we should never have this case?).
Anyways, as an afterthought I figured maybe it would be better to return 'not found' if either the root.json _or_ the repoId is not found, since this method is returning both of them now. WDYT? It's here: https://github.com/advancedtelematic/ota-tuf/pull/235.

Reply:
I don't think that can ever happen... if it happens then that should change on the server side...

If the header is not present that is a server error, and we should not ignore it. If we need to handle it with more logic that what we are doing we would need a better return type for `fetchRoot` that would describe what happened, and users of the library could decide.